perceval/converters/myqlm_converter.py

```diff
@@ -0,0 +1,151 @@
+        self._two_phase_component = Circuit(2) // (0, comp.PS(P("phi1"))) // (1, comp.PS(P("phi2")))
+
+    def convert(self, qlmc, use_postselection: bool = True) -> Processor:
+        r"""Convert a myQLM quantum circuit into a perceval `Processor`.
+
+        :param qlmc: quantum gate-based myqlm circuit
+        :type qlmc: qat.core.Circuit
+        :param use_postselection: when True, uses a `post-processed CNOT` as the last gate. Otherwise, uses only
+            `heralded CNOT`
+        :return: the converted Processor
+        """
+        import qat
+        from qat.core.circuit_builder.matrix_util import circ_to_np
+        # importing the quantum toolbox of myqlm
+        # this nested import fixes automatic class reference generation
+
+        # count the number of CNOT gates to use during the conversion, will give us the number of herald to handle
+        n_cnot = qlmc.count("CNOT")
+        cnot_idx = 0
+
+        n_moi = qlmc.nbqbits * 2  # number of modes of interest = 2 * number of qbits
+        input_list = [0] * n_moi
+        p = Processor(self._backend_name, n_moi, self._source)
+
+        for i in range(qlmc.nbqbits):
+            p.add_port(i * 2, Port(Encoding.DUAL_RAIL, f'Q{i}'))
+            input_list[i * 2] = 1
+        default_input_state = BasicState(input_list)
+
+        for i, instruction in enumerate(qlmc.iterate_simple()):
+            instruction_name = instruction[0]  # name of the Gate
+            instruction_qbit = instruction[-1]  # tuple with list of qbit positions
+            # information carried by instruction
+            # tuple ('Name', [value of the parameter for gate], [list of qbit positions where gate is applied])
+
+            assert instruction_name in qlmc.gate_set, "cannot convert (%s)" % instruction_name
+            # only gates are converted -> checking if instruction is in gate_set of AQASM
+
+            if len(instruction_qbit) == 1:
+                if instruction_name == "H":
+                    ins = Circuit(2, name='H') // BS.H()
+                elif instruction_name == "PH":
+                    phi = instruction[1][0]  # value of the variable parameter in gate
+                    ins = Circuit(2, name='PS') // (1, PS(phi))  # apply phase shift on 2nd mode
+                else:
+                    gate_id = qlmc.ops[i].gate
+                    gate_matrix = qlmc.gateDic[gate_id].matrix  # gate matrix data
+                    gate_u = circ_to_np(gate_matrix)  # gate matrix to numpy
+                    ins = self._create_one_qubit_gate(gate_u)
+                p.add(instruction_qbit[0]*2, ins.copy())
+            else:
+                # only 2 qubit gates
+                c_idx = instruction_qbit[0] * 2
+                c_data = instruction_qbit[1] * 2
+                c_first = min(c_idx, c_data)  # used in SWAP
+
+                if instruction_name == "CNOT":
+                    cnot_idx += 1
```

Comment:
You should declare `mode_map` (`mode_map = None`) and `cnot_processor` (`cnot_processor = None`) before the if statement line 112
Same reason that `ins` above

EDIT: `mode_map` doesn't need to be instantiate if you consider my two next comments